yarnspawner/singleuser.py

```diff
@@ -1,21 +1,34 @@
+import json
 import os
 
-from jupyterhub.singleuser import SingleUserNotebookApp
+from jupyter_server.serverapp import ServerApp
 from jupyterhub.utils import random_port, url_path_join
 from traitlets import default
 
+if not os.environ.get("JUPYTERHUB_SINGLEUSER_APP"):
+    # setting this env prior to import of jupyterhub.singleuser avoids unnecessary import of notebook
+    os.environ["JUPYTERHUB_SINGLEUSER_APP"] = "jupyter_server.serverapp.ServerApp"
+
+try:
+    from jupyterhub.singleuser.mixins import make_singleuser_app
+except ImportError:
+    # backward-compat with jupyterhub < 1.3
```

Comment:
What the point if you've added `'jupyterhub>=4.0.0'`?

Reply:
Good point, let me remove this part.